components/logins/src/migrate_sqlcipher_db.rs

```diff
@@ -67,71 +150,532 @@ fn sqlcipher_3_compat(conn: &Connection) -> Result<()> {
+                        //guid: key,
+                        local_login: Some(l_login),
+                        mirror_login: None,
+                        status: MigrationStatus::Processing,
+                    });
+            }
+            Err(e) => {
+                // We should probably just skip if we can't successfully fetch the row
+                log::warn!("Error getting record from DB: {:?}", e);
+            }
+        }
+    }
+    // Process mirror logins and add to MigrationPlan
+    let mut mirror_stmt = cipher_conn.prepare("SELECT * FROM loginsM")?;
+    let mut mirror_rows = mirror_stmt.query(NO_PARAMS)?;
+    while let Some(row) = mirror_rows.next()? {
+        match get_login_from_row(row) {
+            Ok(login) => {
+                let m_login = MirrorLogin {
+                    login: login.encrypt(&encryptor)?,
+                    server_modified: ServerTimestamp(
+                        row.get::<_, i64>("server_modified").unwrap_or_default(),
+                    ),
+                    is_overridden: row.get("is_overridden").unwrap_or_default(),
+                };
+
+                let key = m_login.login.record.id.clone();
+                migration_plan
+                    .logins
+                    .entry(key)
+                    .and_modify(|l| l.mirror_login = Some(m_login.clone()))
+                    .or_insert(MigrationLogin {
+                        //guid: key,
+                        local_login: None,
+                        mirror_login: Some(m_login),
+                        status: MigrationStatus::Processing,
+                    });
+            }
+            Err(e) => {
+                // We should probably just skip if we can't successfully fetch the row
+                log::warn!("Error getting record from DB: {:?}", e);
+            }
+        }
+    }
+    Ok(migration_plan)
+}
+
+fn migrate_logins(
+    migration_plan: &MigrationPlan,
+    store: &LoginStore,
+    encdec: &EncryptorDecryptor,
+) -> Result<MigrationMetrics> {
+    let import_start = Instant::now();
+    let import_start_total_logins: u64 = migration_plan.logins.len() as u64;
+    let mut num_failed_insert: u64 = 0;
+    let mut insert_errors: Vec<String> = Vec::new();
+
+    for (_, login) in &migration_plan.logins {
+        // Migrate local login first
+        if let Some(local_login) = &login.local_login {
+            match migrate_local_login(store, local_login, &encdec) {
+                Ok(_) => {
+                    if let Some(mirror_login) = &login.mirror_login {
+                        // If successful, then migrate mirror
+                        migrate_mirror_login(&store, &mirror_login)?;
+                    }
+                }
+                Err(e) => {
+                    // If not successful on local, but we have a mirror
+                    // If there is an override -> delete skip both
+                    // IF no override -> attempt to import as normal
```

Comment:
This makes me think that the fixup should be a separate pass and `fixup_mismatched_records` should be last, so the code to handle the `is_overridden=true` but no local record is all in 1 place.  If needed we could collect some data in `MigrationLogin` to distinguish the different cases.  Maybe `local` wouldn't be an `Option`, it would be an enum with several cases:
  - `NotFound`
  - `Invalid`
  - `Valid(Login)`

But that's only if we actually need different behavior for those cases.